std/container/array.d

```diff
@@ -315,7 +315,23 @@ if (!is(Unqual!T == bool))
                 const nbytes = mulu(newLength, T.sizeof, overflow);
                 if (overflow)
                     assert(0);
-                _payload = (cast(T*) realloc(_payload.ptr, nbytes))[0 .. newLength];
+
+                static if (hasIndirections!T)
+                {
+                    auto newPayload = enforce(cast(T*) malloc(nbytes))[0 .. newLength];
```

Comment:
Also, the call to enforce means that setting the length cannot be `nothrow` or `@nogc` when `T` has indirections. Please change this to what `reserve` does, i.e.:
```
auto newPayloadPtr = cast(T*) malloc(sz);
newPayloadPtr || assert(false, "std.container.Array.reserve failed to allocate memory");
```